src/crepe/api/LoopManager.cpp

```diff
@@ -20,51 +22,47 @@ LoopManager::LoopManager() {
 	this->load_system<RenderSystem>();
 	this->load_system<ScriptSystem>();
 	this->load_system<InputSystem>();
+	this->event_manager.subscribe<ShutDownEvent>(
+		[this](const ShutDownEvent & event) { return this->on_shutdown(event); });
 }
 
-void LoopManager::process_input() { this->get_system<InputSystem>().update(); }
+void LoopManager::process_input() {
```

Comment:
waarom is deze functie toegevoegd?

Is het misschien mogelijk om de loop met een fixed_update en frame_update te maken?
Dus 3 functies en dan alles daarinzetten. Nu is het onleesbaar

Reply:
de process_input functie is er altijd al geweest om de delen die te maken hebben met het verwerken van de input te bundelen. ik heb alleen de dispatch hierbij gezet

Reply:
Mij maakt het niet uit wat er is of was geweest. 
Kun je het leesbaarder maken door minder functies?

Reply:
is het niet juist leesbaarder door de process_input functie? want hierdoor staan de functies die te maken hebben met de input bij elkaar in plaats van door de fixed_loop system update calls

Reply:
Door `InputSystem::update` aan te roepen in `process_input` is het niet meer snel te lezen of de `InputSystem` in de fixed of frame loop geupdatet wordt. Het zelfde geldt voor de `render` functie (samenvoegen met `frame_update`